### PR TITLE
feat: add Dockerfiles for development purposes

### PR DESCRIPTION
**Pull request by HarikrishnanBalagopal**
Signed-off-by: Harikrishnan Balagopal <harikrishmenon@gmail.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM quay.io/hari_balagopal/move2kube-website-base:v0.3.5
```
Comment (ashokponkumar) [inline Dockerfile:1]: We can put it either in Move2Kube or Konveyor orgs in quay.

File: Dockerfile.base [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# base image built from https://github.com/just-the-docs/just-the-docs/blob/main/Dockerfile
+FROM quay.io/hari_balagopal/just-the-docs:v1
```
Comment (ashokponkumar) [inline Dockerfile.base:2]: We can put it either in Move2Kube or Konveyor orgs in quay.

**Review by ashokponkumar — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (HarikrishnanBalagopal) [reply Dockerfile:1]: sure, can you add it? don't have the perms

File: Dockerfile.base (hunk 2 above)
Comment (HarikrishnanBalagopal) [reply Dockerfile.base:2]: sure, can you add it? don't have the perms

[the remark this answers is not in the document]
Comment (ashokponkumar) [reply Dockerfile.base:2]: we can make this image to autobuild using github actions whenever the Gemfile or the Dockerfile changes and pus the image to the registry. Can we also move the yarn build here?

[the remark this answers is not in the document]
Comment (HarikrishnanBalagopal) [reply Dockerfile.base:2]: ```COPY Gemfile just-the-docs.gemspec ./```
This line will fail. Those 2 files are present in the just-the-docs repo https://github.com/just-the-docs/just-the-docs

File: Dockerfile.base [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# base image built from https://github.com/just-the-docs/just-the-docs/blob/main/Dockerfile
+FROM quay.io/hari_balagopal/just-the-docs:v0.4.0.rc3
```
Comment (ashokponkumar) [inline Dockerfile.base:2]:
```suggestion
FROM quay.io/konveyor/just-the-docs:v0.4.0.rc3
```

**Review by ashokponkumar — CHANGES_REQUESTED** [review]
Do we also want to add GitHub actions to auto push the images? We can do it along with the versioning of the website too.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+WORKDIR /app
+COPY Gemfile Gemfile.lock ./
+RUN bundle install
+COPY . .
+RUN yarn run build && yarn run bundle
+EXPOSE 4000
+CMD [ "bundle", "exec", "jekyll", "serve", "--host=0.0.0.0"]
```
[the remark this answers is not in the document]
Comment (HarikrishnanBalagopal) [reply Dockerfile]: added the `--incremental`

**Comment by HarikrishnanBalagopal** [comment]
> Do we also want to add GitHub actions to auto push the images? We can do it along with the versioning of the website too.

Not yet, the versioning requires a more substantial over haul of the website and possibly the theme itself.
See https://github.com/just-the-docs/just-the-docs/issues/728

**Review by ashokponkumar — APPROVED** [review]

**Pull request merged**